test/test_torch.py

```diff
@@ -2890,6 +2891,35 @@ def add_neg_dim_tests():
         assert not hasattr(AbstractTestCases._TestTorchMixin, test_name), "Duplicated test name: " + test_name
         setattr(AbstractTestCases._TestTorchMixin, test_name, make_neg_dim_test(name, tensor_arg, arg_constr, types, extra_dim))
 
+# Tests Vital Signs for Torch
+class TestVitalSigns(TestCase):
+    def test_basic_vitals(self):
+        # Protect current environment variable, if set
+        stash = None
+        if 'TORCH_VITAL' in os.environ:
+            stash = os.environ['TORCH_VITAL']
+        os.environ['TORCH_VITAL'] = ''
+        self.assertFalse(torch.vitals_enabled())
+        os.environ['TORCH_VITAL'] = 'ON'
+        self.assertTrue(torch.vitals_enabled())
+
+        # Restore previous state before our test
+        if stash:
+            os.environ['TORCH_VITAL'] = stash
+
+    def test_write_vital(self):
+        stash = None
+        if 'TORCH_VITAL' in os.environ:
+            stash = os.environ['TORCH_VITAL']
+        os.environ['TORCH_VITAL'] = 'ON'
+        self.assertTrue(torch.vitals_enabled())
+        # Ensure we don't segfault a basic call, since we don't have a GET API, it is harder to check.
+        self.assertTrue(torch.set_vital('Dataloader', 'basic_unit_test', 'TEST'))
```

Comment:
is it possible to test output from python? If not, that's fine, there's cpp test, but please leave a more detailed comment.

Reply:
Currently we don't have a way to test output; and if we added one it could only read a subset of vitals (due to our declare vitals anywhere as a variable approach). I added a comment describing the issue and fixed the existing one. Thanks!